safe_eth/eth/clients/etherscan_client_v2.py

```diff
@@ -78,3 +84,81 @@ def is_supported_network(cls, network: EthereumNetwork) -> bool:
         return any(
             item.get("chainid") == str(network.value) for item in supported_networks
         )
+
+
+class AsyncEtherscanClientV2(EtherscanClientV2):
+    def __init__(
+        self,
+        network: EthereumNetwork,
+        api_key: Optional[str] = None,
+        request_timeout: int = int(
+            os.environ.get("ETHERSCAN_CLIENT_REQUEST_TIMEOUT", 10)
+        ),
+        max_requests: int = int(os.environ.get("ETHERSCAN_CLIENT_MAX_REQUESTS", 100)),
+    ):
+        super().__init__(network, api_key, request_timeout)
+        self.async_session = aiohttp.ClientSession(
+            connector=aiohttp.TCPConnector(limit_per_host=max_requests)
+        )
+
+    async def _async_do_request(
+        self, url: str
+    ) -> Optional[Union[Dict[str, Any], List[Any], str]]:
+        """
+        Async version of _do_request
+        """
+        async with self.async_session.get(
+            url, timeout=self.request_timeout
+        ) as response:
+            if response.ok:
+                response_json = await response.json()
+                result = response_json["result"]
+                if "Max rate limit reached" in result:
+                    # Max rate limit reached, please use API Key for higher rate limit
+                    raise EtherscanRateLimitError
+                if response_json["status"] == "1":
+                    return result
+            return None
+
+    async def async_get_contract_source_code(
+        self,
+        contract_address: str,
+    ):
+        """
+        Asynchronous version of get_contract_source_code
+        Does not implement retries
+
+        :param contract_address:
+        """
+        url = self.build_url(
+            f"module=contract&action=getsourcecode&address={contract_address}"
+        )
+        response = await self._async_do_request(url)  # Returns a list
+        return self.process_response(response)
+
+    async def async_get_contract_metadata(
+        self, contract_address: str
+    ) -> Optional[ContractMetadata]:
+        contract_source_code = await self.async_get_contract_source_code(
+            contract_address
+        )
+        if contract_source_code:
```

Comment:
Would it make sense to add this processing in a private method of the superclass and call it here?

Same for the other similar cases in other clients.

WDYT?